### ISSUE TITLE
ELU stucks

### ISSUE THREAD

**Issue by Sebubu** [fchollet/keras#1088]
My model gets stuck as soon as i use ELU. PReLU works without a problem.
I've tried to train it with adadelta and sgd. I've trained it about 60 epochs during the night. Same results.

```python
batch_size = 128
nb_classes = 2
nb_epoch = 500

# input image dimensions
img_rows, img_cols = 50, 50
# number of convolutional filters to use
nb_filters1 = 64
nb_filters2 = 128
nb_filters3 = 256
# size of pooling area for max pooling
nb_pool = 2
# convolution kernel size
nb_conv = 3
#image is rgb
img_channels = 3

#Lamda for L2 regularization
lmda = 5e-4


print("put convnet together")
model = Sequential()

model.add(Convolution2D(nb_filters1, nb_conv, nb_conv, border_mode='full', input_shape=(img_channels, img_rows, img_cols)))
model.add(ELU())
model.add(Dropout(0.2))
model.add(Convolution2D(nb_filters1, nb_conv, nb_conv))
model.add(ELU())
model.add(MaxPooling2D(pool_size=(nb_pool, nb_pool)))

model.add(Dropout(0.2))
model.add(Convolution2D(nb_filters2, nb_conv, nb_conv))
model.add(ELU())
model.add(Dropout(0.2))
model.add(Convolution2D(nb_filters2, nb_conv, nb_conv))
model.add(ELU())
model.add(MaxPooling2D(pool_size=(nb_pool, nb_pool)))

model.add(Dropout(0.2))
model.add(Convolution2D(nb_filters3, nb_conv, nb_conv))
model.add(ELU())
model.add(Dropout(0.2))
model.add(Convolution2D(nb_filters3, nb_conv, nb_conv))
model.add(ELU())
model.add(MaxPooling2D(pool_size=(nb_pool, nb_pool)))

model.add(Flatten())

model.add(Dropout(0.5))
model.add(Dense(2048, W_regularizer=l2(lmda)))
model.add(ELU())

model.add(Dropout(0.5))
model.add(Dense(1024, W_regularizer=l2(lmda)))
model.add(ELU())

model.add(Dropout(0.5))
model.add(Dense(nb_classes, W_regularizer=l2(lmda)))
model.add(Activation('softmax'))

sgd = SGD(lr=0.1, decay=1e-6, momentum=0.9, nesterov=True)
model.compile(loss='categorical_crossentropy', optimizer=sgd)

model.fit(X_train, Y_train, batch_size=batch_size, nb_epoch=nb_epoch, show_accuracy=True, verbose=1, validation_data=(X_test, Y_test))
```
![asdfas](http://s3.postimg.org/4mj06oltf/ELU_stucks.png)
train_loss decreases until 2.0 and remains there. Train_acc, test_loss and test_acc remain on the same level.

Theano and keras has been updated.
Pull Request: #1079

Anyone with the same problem?

**Comment by Sebubu** [comment]
It doesn't like me^^
![elu2](https://cloud.githubusercontent.com/assets/8782386/11424996/e339c7e4-9451-11e5-9ca0-468b45720e9b.png)

**Comment by jfsantos** [comment]
The ELU implementation proposed in that PR has not yet been approved, so I think it's better to discuss this in #1079 instead of opening a new issue. This is not (currently) an issue with Keras, as this code is not part of Keras yet :)

**Comment by fchollet** [comment]
Try plotting the response of that ELU layer against floats between -10 and +10. Then we'll see if it matches the paper.

**Comment by Sebubu** [comment]
- Plot and derivation seem ok and equal to the paper. 
- Elu function #1083 has the same "issue" like the implemented.
- The model above works if dropout is replaced by batchnorm
- mnist_cnn.py example works with dropout and elu

Maybe just a producible coincidence. I don't get it. 

The issue is done for me so far. Nevertheless I believe it was worth to be documented.

Thank you for support :)
